lib/ansible/modules/cloud/docker/docker_swarm_service.py

```diff
@@ -615,9 +615,11 @@ def from_ansible_params(ap, old_service):
         s.command = ap['command']
         if isinstance(s.command, string_types):
             s.command = shlex.split(s.command)
-        elif isinstance(s.command, list) and all(
-            isinstance(item, string_types) for item in s.command
-        ):
+        elif isinstance(s.command, list):
+            if not all(isinstance(item, string_types) for item in s.command):
```

Comment:
It is probably better to do this as a loop, so you know the index of the (first) item where it is not a string. That allows to produce a more helpful error message. YAML has some tricky conversions (like `'off' --> False` or `'n' --> False`) which are hard to detect if you don't know about them.

Reply:
Updated the error message to be more detailed and added tests.